apis/supernova/src/hooks.js

```diff
@@ -990,6 +1003,28 @@ export function onTakeSnapshot(cb) {
   h.fn = cb;
 }
 
+/**
+ * Registers a callback that is called when a context menu item is added.
+ * @entry
+ * @param {function(menu, event): Promise<menu>} addItemCallback
```

Comment:
why do the a callback return Promise<menu> and not Promise<undefined>?
(It feels just like an unnecessary complication)

and why this API (why a callback)?
and not a object describing a single menu item matching how `useAction` works

Reply:
@Caele suggested to implement it similar to `setSnapshotData`, so we followed the same pattern. You pass a callback where you can do whatever you need. As I understand it this might be revisited when there is a context menu implementation for nebula, right @Caele?

Reply:
@Caele, Are we right ?

Reply:
@T-Wizard as Max explained here menu is returned since we are implementing this function in sense-client and now the new item (CopyCellValue) has been added to the list of context menu items.

Reply:
I can see the reason to use a callback bacause it is similar to how it works in sense-client (and maybe becasue it is similar to setSnapshotData).
(I still think we should take some time to consider the api some more to reduce the need to change it later)

the return values is a separate point and I don't think the callback should br required to return the menu.
(I think it will always be the first argument as I don't see the point of returning it)
returning a promise could be relevant if the callback should be able to do something that takes time.

Reply:
Do you think we should apply this consideration in the current PR? if so then I should discuss it with @haxxmaxx  and @Caele.

Yes. It is already fixed, `export function onContextMenu(addItemCallback: (menu: Menu, event: any) => menu): void;` , I think it is what you meant, (just a small miss spell I found, I'm going to fix it).

Reply:
no I meant to change it to `export function onContextMenu(addItemCallback: (menu: Menu, event: any) => Promise<void>): void;`

Reply:
Ok sorry I misunderstood. Umm.. event then I don't think Promise is needed, there is nothing that takes time to do it. I'll try return void and will see.